toolchain/sem_ir/facet_type_info.cpp

```diff
@@ -39,42 +39,55 @@ static auto RequiredLess(const IdentifiedFacetType::RequiredInterface& lhs,
 static auto SubtractSorted(
     llvm::SmallVector<FacetTypeInfo::ImplsConstraint>& a,
     const llvm::SmallVector<FacetTypeInfo::ImplsConstraint>& b) -> void {
-  auto a_iter = a.begin();
-  auto b_iter = b.begin();
-  auto a_end = a.end();
-  auto b_end = b.end();
+  typedef llvm::SmallVector<FacetTypeInfo::ImplsConstraint>::iterator Iter;
+  typedef llvm::SmallVector<FacetTypeInfo::ImplsConstraint>::const_iterator
+      ConstIter;
+  Iter a_iter = a.begin();
+  ConstIter b_iter = b.begin();
```

Comment:
I think it's preferable to just use `Iter` for both here; otherwise the comparison make a type conversion?

Reply:
Rearranged so the code is clearer.